File: Task/Convert-seconds-to-compound-duration/Java/convert-seconds-to-compound-duration-2.java
public class CompoundDuration {

    public static void main(String[] args) {
        compound(7259);
        compound(86400);
        compound(6000_000);
    }

    private static void compound(long seconds) {
        StringBuilder sb = new StringBuilder();

        seconds = addUnit(sb, seconds, 604800, " wk, ");
        seconds = addUnit(sb, seconds, 86400, " d, ");
        seconds = addUnit(sb, seconds, 3600, " hr, ");
        seconds = addUnit(sb, seconds, 60, " min, ");
        addUnit(sb, seconds, 1, " sec, ");

        sb.setLength(sb.length() > 2 ? sb.length() - 2 : 0);

        System.out.println(sb);
    }

    private static long addUnit(StringBuilder sb, long sec, long unit, String s) {
        long n;
        if ((n = sec / unit) > 0) {
            sb.append(n).append(s);
            sec %= (n * unit);
        }
        return sec;
    }
}

File: Task/Convert-seconds-to-compound-duration/JavaScript/convert-seconds-to-compound-duration-2.js
(() => {
    "use strict";

    // ---------------- COMPOUND DURATION ----------------

    // compoundDuration :: [String] -> Int -> String
    const compoundDuration = labels =>
        nSeconds => weekParts(nSeconds)
        .map((v, i) => [v, labels[i]])
        .reduce((a, x) =>
            a.concat(
                x[0] ? [
                    `${x[0]} ${x[1] || "?"}`
                ] : []
            ), []
        )
        .join(", ");


    // weekParts :: Int -> [Int]
    const weekParts = nSeconds => [0, 7, 24, 60, 60]
        .reduceRight((a, x) => {
            const
                r = a[0],
                mod = x !== 0 ? r % x : r;

            return [
                (r - mod) / (x || 1),
                [mod, ...a[1]]
            ];
        }, [nSeconds, []])[1];


    // ---------------------- TEST -----------------------
    // main :: IO ()
    const main = () => {
        const localNames = ["wk", "d", "hr", "min", "sec"];

        return [7259, 86400, 6E6]
            .map(nSeconds =>
                `${nSeconds} -> ${
                compoundDuration(localNames)(nSeconds)
            }`).join("\n");
    };

    // MAIN ---
    return main();
})();
